text/0000-project-groups.md

```diff
@@ -0,0 +1,391 @@
+  within the context of a Rust team
+- Project groups are created via an RFC and have a "parent team" (or
+  multiple teams)
+- The groups then drive the project to completion, e.g. by authoring follow-up
+  RFCs and doing design work.
+- Each project group typically has:
+  - A charter outlining the group's scope and goals.
+  - Appointed shepherds and team liasons.
+  - An associated repository.
+  - Regular meetings.
+  - Dedicated streams on Discord/Zulip/etc.
+- Define working groups to refer to the "domain working groups" that are created
+  to explore particular domains, such as embedded, CLI, etc.
+  - They have a charter and defined leads but operate more independently from
+    the Rust teams.
+- Define community group as the term for groups not formally affiliated with the
+  Rust project.
+
+# Motivation
+
+Working groups were not created through the RFC process, as such there's not
+much documentation on exactly what a working group does, what are its
+responsibilities, and also importantly setting expectations or goals. There is
+one definition of "Working Group" available in [rust-lang#54445]; quoted below.
+
+> A Rust Working Group is a set of people working at common purpose. Working
+> Groups are associated with a Rust team. Unlike a Rust Team, Working Groups
+> don't have "formal decision making power", though often they are charged with
+> drawing up recommendations, RFCs, or other documents for the teams (which is
+> then intended to make the final decision).
+
+While this definition is true in the broad sense, it does not define the
+difference between what has come to be called "Domain Working Groups" and
+"Team Working Groups". Additionally the community has also adopted the Working
+Group terminology for organising and working on areas they are interested in,
+independent of The Rust Programming Language's Working Groups. It's great that
+we have a community able to self organise in this way, however it has led to
+some confusion over who is supporting these efforts, and whether they're
+considered _"official"_ Working Groups.
+
+This RFC aims to provide clarity by providing new distinct terminology that
+matches the intent behind each of these three separate groups, as well as codify
+the processes that these groups have been using to help facilitate creating
+new groups.
+
+[rust-lang#54445]: https://github.com/rust-lang/rust/issues/54445
+
+# Guide-level explanation
+
+To address this confusion this RFC proposes breaking up the single "Working
+Group" term, into three distinct terms.
+
+- **Working Group** would map most directly to the previous "Domain Working
+  Group" terminology, focused on building and organising around new areas of
+  Rust that aren't covered by the current team roster.
+- **Project Group** would replace the "Team Working Group", and would serve as a
+  catch all term for sub teams for the existing teams to organise around
+  specific efforts, such as certain project or effort in the Rust team.
```

Comment:
Well I feel like it being broad is an asset. I feel like the dichotomy between those things is expressed by the parent team. e.g. The lang team's project groups are generally about changing the language, and the release team project groups are about managing and improving Rust's release process.

I also think we shouldn't try to highlight some efforts as being "bigger" or more important than others. Both because that might inadvertently put undue pressure or attention on its members, and I don't think you can reasonably measure. Just because inline assembly is more user visible thing than tagging GitHub issues doesn't mean that the triage-wg doesn't have a big impact on improving the quality of Rust.

I also want to try to be flexible as if a project group changes from being *'short-lived and "goal" driven'* to being something that's more long term, that they don't have to suddenly change all references everywhere from project group to "foo group". (An example being an effort to improve a part of the compiler that ends up producing a useful tool that people want maintain).